engine/errors.go

```diff
@@ -146,3 +146,28 @@ func LogErrorWithMsg(log zerolog.Logger, msg string, err error) {
 	// all other errors should just be logged as usual
 	log.Error().Str("error_type", "internal_error").Err(err).Msg(msg)
 }
+
+// Return the error type and whether the error is internal error
+func ErrorType(err error) (string, bool) {
```

Comment:
I personally think that we shouldn't have this it all, if by some reason one of the functions returns wrong errors it should better crash the node and let developers investigate compared to comprising the network with byzantine behavior because it didn't crash in some other cases. To me defining this set of errors is very error phone even on engine level.